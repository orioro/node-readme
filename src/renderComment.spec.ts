import {
  renderFunctionComment,
  renderDefaultComment
} from './renderComment'

test('renderDefaultComment(comment)', () => {
  expect(renderDefaultComment({
    commentType: 'typedef',
    name: 'SomeSpecialType',
    description: 'Integer finibus mi sed erat condimentum rhoncus id eu ligula. Mauris sed odio orci.',
    properties: [
      { name: 'prop1', type: 'string', description: 'Morbi egestas suscipit pharetra' },
      { name: 'prop2', type: 'boolean', description: 'In maximus est nibh' }
    ]
  }))
  .toMatchSnapshot()
})

test('renderFunctionComment(comment)', () => {
  expect(renderFunctionComment({
    commentType: 'function',
    name: 'someFunction',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    params: [
      {
        name: 'param1',
        type: 'String',
        description: 'Some description'
      },
      {
        name: 'param2',
        type: 'SomeSpecialType',
        description: 'Some description'
      },
      {
        name: 'param3',
        type: 'Object',
        description: 'Some param3 description',
        properties: [
          {
            name: 'key1',
            type: 'Number',
            description: 'Some key1 description'
          },
          {
            name: 'key2',
            type: 'String',
            description: 'Some key2 description'
          }
        ]
      },
      {
        name: 'param4',
        type: 'Object[]',
        description: 'Some description',
        properties: [
          {
            name: 'key1',
            type: 'Number',
            description: 'Some key1 description'
          },
          {
            name: 'key2',
            type: 'String',
            description: 'Some key2 description'
          }
        ]
      }
    ],
    returns: {
      name: 'result',
      type: 'SomeType',
      description: 'Some result description'
    }
  }))
  .toMatchSnapshot()
})

