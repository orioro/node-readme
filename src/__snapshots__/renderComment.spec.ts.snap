// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderDefaultComment(comment) 1`] = `
"##### \`SomeSpecialType\`

Integer finibus mi sed erat condimentum rhoncus id eu ligula. Mauris sed odio orci.

- \`prop1\` {string}
- \`prop2\` {boolean}"
`;

exports[`renderFunctionComment(comment) 1`] = `
"##### \`someFunction(param1, param2, param3, param4)\`

Lorem ipsum dolor sit amet, consectetur adipiscing elit

- \`param1\` {String}
- \`param2\` {SomeSpecialType}
- \`param3\` {Object}
  - \`key1\` {Number}
  - \`key2\` {String}
- \`param4\` {Object[]}
  - \`key1\` {Number}
  - \`key2\` {String}
- Returns: \`result\` {SomeType} Some result description"
`;
